src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,6 +1256,794 @@ subroutine eigencalving(domain, err)
+                    damage(iCell) = damage_downstream / n_damage_downstream
+                end if
+
+            end if
+        end do
+        ! set the damage for a cell at the GL to the average damage value for its neighbor cells downstream on the ice shelf 
+                        
+
+        if (.true.) then
+
+             call mpas_log_write("source value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(source), maxval(source)/))
+             call mpas_log_write("szero  value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(szero), maxval(szero)/))
+             call mpas_log_write("nstar value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(nstar), maxval(nstar)/))
+             call mpas_log_write("deltat value range: Min=$r, Max=$r", &
+                        realArgs=(/deltat, deltat/))
+             call mpas_log_write("sigmamax value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(sigmamax), maxval(sigmamax)/))
+             call mpas_log_write("damage value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(damage), maxval(damage)/))
+             call mpas_log_write("A value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(meanFlowParamA), maxval(meanFlowParamA)/))
+             call mpas_log_write("alpha value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(alpha), maxval(alpha)/))
+        end if
+        ! temporary message output in the log file
+
+
+         call mpas_deallocate_scratch_field(calvingFrontMaskField, .true.)
+         call mpas_deallocate_scratch_field(meanFlowParamAField, .true.)
+         !call mpas_deallocate_scratch_field(viscField, .true.)
+         call mpas_deallocate_scratch_field(sigmaxxField, .true.)
+         call mpas_deallocate_scratch_field(sigmayyField, .true.)
+         call mpas_deallocate_scratch_field(sigmaminField, .true.)
+         !call mpas_deallocate_scratch_field(sigmamaxField, .true.)
+         call mpas_deallocate_scratch_field(alphaField, .true.)
+         !call mpas_deallocate_scratch_field(epsmaxField, .true.)
+         !call mpas_deallocate_scratch_field(nstarField, .true.)
+         !call mpas_deallocate_scratch_field(szeroField, .true.)
+         !call mpas_deallocate_scratch_field(sourceField, .true.)
+         !call mpas_deallocate_scratch_field(ddamagedtField, .true.)
+         call mpas_deallocate_scratch_field(forceField, .true.)
+         call mpas_deallocate_scratch_field(meanTField, .true.)
+         ! call mpas_deallocate_scratch_field(damageField, .true.)
+
+         block => block % next
+             
+      enddo
+
+   end subroutine calculate_damage 
+
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!    routine restore_damage
+!
+!> \author Tong Zhang
+!> \date   Oct. 2019
+!> \details restore the damage to the original value before advection if it exceeds some large number
```

Comment:
Can you add more explanation for why this should happen?

Reply:
an additional sentence is added to make it explicit that it is for controlling damage healing

Reply:
I think maybe the confusion here is in the way it's worded. To me, if you are "restoring damage" to avoid letting healing take place, then that means you are holding the damage at some larger value (or resetting it to that larger value) rather than letting it decrease (heal) as it wants to. 

But in the description line above, you say "restore the damage ... if it exceeds some large number", which to me sounds like you are instead _decreasing_ the damage. 

Possibly what you mean here is that, if the damage exceeds some threshold value, then you do not allow healing to take place?

A related question -- if we want to / need to look at the effects of damage in the absence of healing, this alone does not seem adequate. It seems like looking at that would require us to avoid healing of damage everywhere rather than just in locations where damage exceeds some critical value.

Reply:
Yes, your understanding is correct. I've updated the description a bit so it might look a bit more clear now. For your last question, we have another option, config_damage_restore_threshold, to do that. If we set config_damage_restore_threshold to 0 (which means if damage > 0, we'll restore its old value), then there will be no healing everywhere on the ice shelf.